### ISSUE TITLE
Flutter Engine build fails on Mac/Arm64 host

### ISSUE THREAD

**Issue by knopp** [flutter/flutter#97503]
OpenGL depends on swift_shader, which does not support Arm64 target. This breaks the engine build on Mac/Arm64. Also the `create_arm_gen_snapshot` target should only be used when building for ios.

**Comment by knopp** [comment]
Related: https://github.com/flutter/flutter/issues/84453

**Comment by jmagman** [comment]
Some of this is tracked in https://github.com/flutter/flutter/issues/103386

**Comment by cbracken** [comment]
Agh - missed this when I filed that issue; I've transferred the note about gen_snapshot to that bug; closing to de-duplicate. The bigger issue is SwiftShader deprecating GL support altogether to focus on Vulkan. The recommended path forward that we've been pursuing is using SwiftShader in the "swangle" configuration wherein we make GL calls against ANGLE which translates to Vulkan which we then use SwiftShader to provide a software implementation of.

I did some work on this a bit earlier this year -- our ANGLE dependency was over a year out of date and missing a lot of the support needed to use it in a "swangle" setup. I rolled it to tip of tree about a month ago. I believe @dnfield is picking up the rest. 

Deduplicating this against #90425 for the migration to swangle and #103386 for the overall remaining steps to do a full host build on arm64.
